Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs

```diff
@@ -37,15 +42,51 @@ public class WindowsMixedRealityDeviceManager : BaseInputDeviceManager
             uint priority = DefaultPriority,
             BaseMixedRealityProfile profile = null) : base(registrar, inputSystem, name, priority, profile) { }
 
+        /// <inheritdoc/ >
+        public override bool CheckCapability(MixedRealityInputCapabilities capability)
+        {
+            if (WindowsApiChecker.UniversalApiContractV8_IsAvailable)
+            {
+                // Windows 10 1903 contains an API that allows to check input type support.
+#if WINDOWS_UWP
+                switch (capability)
+                {
+                    case MixedRealityInputCapabilities.ArticulatedHand:
+                    case MixedRealityInputCapabilities.GGVHand:
+                        return WindowsInputSpatial.SpatialInteractionManager.IsSourceKindSupported(WindowsInputSpatial.SpatialInteractionSourceKind.Hand);
+
+                    case MixedRealityInputCapabilities.MotionController:
+                        return WindowsInputSpatial.SpatialInteractionManager.IsSourceKindSupported(WindowsInputSpatial.SpatialInteractionSourceKind.Controller);
+                }
+#endif // WINDOWS_UWP
+            }
+            else
+            {
+                // Pre-Windows 10 1903.
```

Comment:
Nit: Maybe move this to the same line as the else, if only because I'm not sure if this statement modifies the entire else block, or just the if statement right below. If this were at the else location (or if there were even an extra line below this) I think it would less ambiguous